### PR TITLE
Make clear the propagated/non-recording span concept needs no type.

### LINKED ISSUE #1087: Clarify purpose of `PropagatedSpan`
From the discussion on #1086 and as discussed in the SIG spec meeting:
We should clarify the purpose of `PropagatedSpan` since the current description in the API seems to be confusing.

### LINKED ISSUE #1124: Revert use of PropagatedSpan back to using a special Context key for extracted Span Context
**What are you trying to achieve?**

In discussion last night in the Spec SIG meeting we talked about the confusion added with PropagatedSpan and whether it didn't cause more confusion than the issue it was trying to resolve (the code required to check if an extracted Span Context exists if there isn't a parent in the current Context or the Context manually passed as the parent to `StartSpan`.

I'm opening a revert of the PropagatedSpan PR to go along with this issue.

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #1087, closes #1124, closes #1125

## Changes

Makes clear that the non-recording span / propagated Span has no semantic meaning and should not be a new public type.

### CODE REVIEW

File: specification/trace/api.md [hunk 1]
```diff
@@ -637,13 +637,16 @@ timestamps to the Span object:
...
 
-### Propagated Span creation
+### Wrapping a SpanReference in a Span
 
 The API MUST provide an operation for wrapping a `SpanReference` with an object
 implementing the `Span` interface. This is done in order to expose a `SpanReference`
 as a `Span` in operations such as in-process `Span` propagation.
 
-If a new type is required for supporting this operation, it SHOULD be named `PropagatedSpan`.
+If a new type is required for supporting this operation, it SHOULD not be exposed
```
Comment (bogdandrutu) [inline specification/trace/api.md:646]: Maybe also clarify what @tsloughter suggested, the returned wrapped Span has `is_recording` return always false

**Review by bogdandrutu — APPROVED** [review]

File: specification/trace/api.md (hunk 1 above)
Comment (Oberon00) [reply specification/trace/api.md:646]: This is already written below:

> The behavior is defined as follows:
> ...
> - `IsRecording` MUST return `false` to signal that events, attributes and other elements
>  are not being recorded, i.e. they are being dropped.

Comment (bogdandrutu) [reply specification/trace/api.md:646]: Now I am confused about the discussion in the issue, need to re-read.

Comment (Oberon00) [reply specification/trace/api.md:646]: I *think* @tsloughter understood that PropagatedSpan was meant to be distinguishable by users like with isinstance checks and wondered why IsRecording is not enough for that.

**Comment by tsloughter** [comment]
I just don't get why any of this is needed. I think I'd be fine with not using a separate context key if none of this was added to the API spec and the spec of the extractor simply said a Span should be created from propagated trace context with `is_recording` as `false`.

**Comment by Oberon00** [comment]
Also, from a performance perspective, a non-recording span can be implemented more efficiently and lightweight than a full span.

**Comment by Oberon00** [comment]
And there is also no other API to create a non-recording span (except if you muck around with samplers maybe).

**Review by carlosalberto — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]
This is much easier for me to understand, thanks

File: specification/trace/api.md [hunk 2]
```diff
@@ -637,13 +637,16 @@ timestamps to the Span object:
...
+publicly if possible (e.g. by only exposing a function that returns something
+with the Span interface type). If a new type is required to be publicly exposed
```
Comment (arminru) [inline specification/trace/api.md:648]:
```suggestion
with the Span interface type). If a new type is required to be publicly exposed,
```

**Review by arminru — APPROVED** [review]

**Comment by carlosalberto** [comment]
@open-telemetry/specs-approvers Please review. The more eyes/approvals, the better ;)

**Comment by bogdandrutu** [comment]
@yurishkuro @SergeyKanzhelev ?

**Comment by carlosalberto** [comment]
Ping @SergeyKanzhelev @yurishkuro

**Review by yurishkuro — APPROVED** [review]

**Review by SergeyKanzhelev — APPROVED** [review]

**Pull request merged**